api/apiFunctions.ts

```diff
@@ -301,3 +297,100 @@ export async function createEntry({
     throw new Error('Error creating entry');
   }
 }
+
+/**
+ * Retrieves a list of all leagues.
+ * @returns {ILeague[]} A list of all available leagues.
+ */
+export async function getAllLeagues(): Promise<IAllLeagues[]> {
+  try {
+    const response = await databases.listDocuments(
+      appwriteConfig.databaseId,
+      Collection.LEAGUE,
+    );
+
+    const allLeagues = response.documents.map((league) => ({
+      leagueId: league.$id,
+      leagueName: league.leagueName,
+      logo: '',
+      participants: league.participants,
+      survivors: league.survivors,
+    }));
+
+    return allLeagues;
+  } catch (error) {
+    console.error(error);
+    throw new Error('Error getting all leagues');
+  }
+}
+/**
+ * Retrieves the user document ID dynamically from the database.
+ * @param userId - The user ID
+ * @returns {string} - The user document ID
+ */
+export const getUserDocumentId = async (userId: string): Promise<string> => {
+  try {
+    const response = await databases.listDocuments(
+      appwriteConfig.databaseId,
+      Collection.USERS,
+      [Query.equal('userId', userId)],
+    );
+
+    if (response.documents.length === 0) {
+      throw new Error('User document not found');
+    }
+
+    // Assuming only one document per userId
+    return response.documents[0].$id;
+  } catch (error) {
+    console.error('Error getting user document ID:', error);
+    throw new Error('Error getting user document ID');
+  }
+};
+
+/**
+ * Adds a user to a league by updating the user's entry document.
+ * @param {string} userId - The ID of the user to add to the league.
+ * @param {string} leagueId - The ID of the league to add the user to.
+ * @param updatedData
+ * @returns {Promise<void>} A promise that resolves when the user has been added to the league.
+ */
+export async function addUserToLeague({
+  userId,
+  selectedLeague,
+  selectedLeagues,
+  participants,
+  survivors,
+}: {
+  userId: string;
+  selectedLeague: string;
+  selectedLeagues: string[];
+  participants: string[];
+  survivors: string[];
+}): Promise<void> {
+  try {
+    const documentId = await getUserDocumentId(userId);
+
+    await databases.updateDocument(
+      appwriteConfig.databaseId,
+      Collection.USERS,
+      documentId,
+      {
+        leagues: selectedLeagues,
+      },
+    );
+
+    await databases.updateDocument(
+      appwriteConfig.databaseId,
+      Collection.LEAGUE,
+      selectedLeague,
+      {
+        participants: participants,
+        survivors: survivors,
+      },
+    );
+  } catch (error) {
+    console.error('Error adding user to league', error);
+    throw new Error('Error adding user to league');
```

Comment:
Add `error` prop to this error for the stack trace.

Reply:
updated